pkg/serviceproxy/service_proxy.go

```diff
@@ -67,30 +67,48 @@ func (s *serviceProxy) AddFlags(cmd *cobra.Command) {
 
 func (s *serviceProxy) Run(ctx context.Context) error {
 	var err error
+	customChecks := []healthz.Checker{}
 
 	if err := s.validate(); err != nil {
 		return err
 	}
 
-	// add configchecker into http probes
-	cc, err := addonutils.NewConfigChecker("http-service-proxy", s.cert, s.key, s.apiserverCA, s.ocpserviceCA)
+	// get root CAs
+	s.rootCAs = x509.NewCertPool()
+	// ca for accessing apiserver
+	const (
+		rootCAFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	)
+	apiserverPem, err := ioutil.ReadFile(rootCAFile)
 	if err != nil {
 		klog.Fatal(err)
 	}
+	s.rootCAs.AppendCertsFromPEM(apiserverPem)
+	// ca for accessing ocp services
+	ocpserviceCAPem, err := ioutil.ReadFile(s.ocpserviceCA)
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.Infof("ocpservice-ca is not provided")
+		} else {
+			klog.Fatal(err)
+		}
+	} else {
+		s.rootCAs.AppendCertsFromPEM(ocpserviceCAPem)
+
+		// add configchecker into http probes only when ocpservice-ca is provided
+		cc, err := addonutils.NewConfigChecker("http-service-proxy", s.cert, s.key, s.ocpserviceCA)
```

Comment:
should you always add probe, since cert/key can be rotated, and also ca for kube-api should be considered?

Reply:
Sure, if we consider kube-apiserver in, then this should always be added.

Reply:
Fixed.